mill-build/src/millbuild/MillJavaModule.scala

```diff
@@ -29,13 +31,55 @@ trait MillJavaModule extends JavaModule {
   def localTestOverridePaths =
     Task { upstreamAssemblyClasspath() ++ Seq(compile().classes) ++ resources() }
 
-  def transitiveLocalTestOverrides: T[Map[String, String]] = Task {
-    val upstream = Task.traverse(moduleDeps ++ compileModuleDeps) {
-      case m: MillJavaModule => m.transitiveLocalTestOverrides.map(Some(_))
-      case _ => Task.Anon(None)
-    }().flatten.flatten
-    val current = Seq(localTestOverride())
-    upstream.toMap ++ current
+  def localTestExtraModules: Seq[MillJavaModule] = Nil
+  def localTestRepositories: T[Seq[PathRef]] = {
+
+    def recursive[T](start: T, deps: T => Seq[T]): Seq[T] = {
```

Comment:
I find this code surprisingly confusing. What is going on here? Is it just a depth-first or breadth-first search? If so, could we implement it with a more traditional while-loop and mutable queue/stack?